src/data/polynomial/bernstein.lean

```diff
@@ -0,0 +1,279 @@
+begin
+  dsimp [bernstein_polynomial],
+  simp [polynomial.derivative_pow],
+end
+
+lemma iterate_derivative_at_zero_eq_zero_of_lt (n : ℕ) {ν k : ℕ} :
+  k < ν → (polynomial.derivative^[k] (bernstein_polynomial n ν)).eval 0 = 0 :=
+begin
+  cases ν,
+  { rintro ⟨⟩, },
+  { intro w,
+    replace w := nat.lt_succ_iff.mp w,
+    revert w,
+    induction k with k ih generalizing n ν,
+    { simp [eval_at_0], rintro ⟨⟩, },
+    { simp only [derivative_succ, int.coe_nat_eq_zero, int.nat_cast_eq_coe_nat, mul_eq_zero,
+        function.comp_app, function.iterate_succ,
+        polynomial.iterate_derivative_sub, polynomial.iterate_derivative_cast_nat_mul,
+        polynomial.eval_mul, polynomial.eval_nat_cast, polynomial.eval_sub],
+      intro h,
+      right,
+      rw ih,
+      simp only [sub_zero],
+      convert @ih (n-1) (ν-1) _,
+      { omega, },
+      { omega, },
+      { exact le_of_lt h, }, }, },
+end
+
+@[simp]
+lemma iterate_derivative_succ_at_zero_eq_zero (n ν : ℕ) :
+  (polynomial.derivative^[ν] (bernstein_polynomial n (ν+1))).eval 0 = 0 :=
+iterate_derivative_at_zero_eq_zero_of_lt n (lt_add_one ν)
+
+@[simp]
+lemma iterate_derivative_at_0 (n ν : ℕ) :
+  (polynomial.derivative^[ν] (bernstein_polynomial n ν)).eval 0 = (falling_factorial n ν : ℕ) :=
+begin
+  induction ν with ν ih generalizing n,
+  { simp [eval_at_0], },
+  { simp [derivative_succ, ih],
+    rw [falling_factorial_eq_mul_left],
+    push_cast, }
+end
+
+lemma iterate_derivative_at_0_ne_zero (n ν : ℕ) (h : ν ≤ n) :
+  (polynomial.derivative^[ν] (bernstein_polynomial n ν)).eval 0 ≠ 0 :=
+begin
+  simp only [int.coe_nat_eq_zero, bernstein_polynomial.iterate_derivative_at_0, ne.def],
+  exact nat.falling_factorial_ne_zero h,
+end
+
+/--
+Rather than redoing the work of evaluating the derivatives at 1,
+we use the symmetry of the Bernstein polynomials.
+-/
+lemma iterate_derivative_at_one_eq_zero_of_lt (n : ℕ) {ν k : ℕ} :
+  k < n - ν → (polynomial.derivative^[k] (bernstein_polynomial n ν)).eval 1 = 0 :=
+begin
+  intro w,
+  rw flip' _ _ (show ν ≤ n, by omega),
+  simp [polynomial.eval_comp, iterate_derivative_at_zero_eq_zero_of_lt n w],
+end
+
+@[simp]
+lemma iterate_derivative_at_1 (n ν : ℕ) (h : ν ≤ n) :
+  (polynomial.derivative^[n-ν] (bernstein_polynomial n ν)).eval 1 =
+    (-1)^(n-ν) * (falling_factorial n (n-ν) : ℕ) :=
+begin
+  rw flip' _ _ h,
+  simp [polynomial.eval_comp],
+end
+
+lemma iterate_derivative_at_1_ne_zero (n ν : ℕ) (h : ν ≤ n) :
+  (polynomial.derivative^[n-ν] (bernstein_polynomial n ν)).eval 1 ≠ 0 :=
+begin
+  simp only [int.coe_nat_eq_zero, bernstein_polynomial.iterate_derivative_at_1 _ _ h, ne.def,
+    false_or, int.coe_nat_eq_zero, int.neg_one_pow_ne_zero, mul_eq_zero],
+  exact nat.falling_factorial_ne_zero (nat.sub_le _ _),
+end
+
+open submodule
+
+lemma linear_independent_aux (n k : ℕ) (h : k ≤ n + 1):
+  linear_independent ℚ (λ ν : fin k, (bernstein_polynomial n ν).map (algebra_map ℤ ℚ)) :=
```

Comment:
For the Chebyshev polynomials, Witt polynomials, and cyclotomic polynomials, we have made the coefficient ring explicit + a simp lemma that always simps away `(some_poly).map some_ring_hom`.
Do you think it's worth it to apply the same strategy here? I like how on average it makes statements cleaner. But maybe in this case you almost always work explicitly with `int`, and this is the only place where you map to other coefficients?

Reply:
At first I was dubious, but it works beautifully.

I will still have to propagate this forward to the other PRs, but I only expect it to help further.

Reply:
Should the same principle be applied to #6309, the Bernoulli polynomnials?